.github/workflows/ci.yml

```diff
@@ -0,0 +1,55 @@
+name: CI
+
+on:
+  pull_request:
+    branches:
+      - main
+
+concurrency:
+  group: ${{ github.ref_name }}-${{ github.workflow }}
+  cancel-in-progress: true
+
+jobs:
+  test_foreman:
+    name: Test with Foreman
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4  
+
+      - name: Clone Foreman
+        run: git clone --branch develop https://github.com/theforeman/foreman.git ../foreman
```

Comment:
It would be better to use `actions/checkout` to create the clone. https://github.com/actions/checkout?tab=readme-ov-file#usage documents the options you can use to achieve the same.

Reply:
Thanks, I've updated it.